### PR TITLE
Use types to select fields for form input

### PR DESCRIPTION
**Pull request by leonidr-da**
@georg-da What do you think?

Not the previous, but my Typescript skills are getting better.

### CODE REVIEW

File: ui/src/pages/report/Report.tsx [hunk 1]
```diff
@@ -7,48 +7,75 @@ import TableBody from "@material-ui/core/TableBody";
...
 
 export default function Report() {
   const classes = useStyles();
 
-  const [ props, setProps ] = useState<InputDialogProps>({ open: false, title: "", fields: [], onClose: async () => {} });
-  const party = useParty();
-  const ledger = useLedger();
-  const assets = useStreamQuery(Asset).contracts;
-  
+  const defaultGive = { newOwner : " " };
```
Comment (GeorgSchneider) [inline ui/src/pages/report/Report.tsx:17]: why the space here?

**Comment by GeorgSchneider** [comment]
I like this a lot for the better type safety. My only concern is that it constrains you to have the full type of a choice (or template) as the state in the dialog. What if I only want the user to fill in parts of the parameters, and some others I compute eg. based on the user input? Then I'd have to put some "dummy" values into the `defaultValue`. Not sure if that's a real concern though

File: ui/src/pages/report/Report.tsx (hunk 1 above)
Comment (leonidr-da) [reply ui/src/pages/report/Report.tsx:17]: That's an error.

**Comment by leonidr-da** [comment]
That's a good idea, let me address it.

**Comment by leonidr-da** [comment]
@georg-da We restrict the date of appraisal to be the current date and limit what `field`s we generate via types.

**Comment by leonidr-da** [comment]
Also ameliorate https://github.com/digital-asset/daml-ui-template/issues/22 😄

**Comment by GeorgSchneider** [comment]
This is really good, I think the template is much more accessible now. I've committed a couple of cosmetic changes, in particular I've come around to using functions instead of lambdas again to make the code easier to understand for beginners. I'll merge this now once the build has passed.

**Comment by leonidr-da** [comment]
Great!

**Pull request merged**